src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -410,24 +410,50 @@ def _query_workspace_with_load_id(
     workspace_id: int,
     load_id: str,
     obj_names: list[str],
-) -> tuple[list[str], list[str]]:
+    assembly_objs_only: bool = True,
+) -> tuple[list[str], list[str], list[str], list[str]]:
     if len(obj_names) > _WS_MAX_BATCH_SIZE:
         raise ValueError(
             f"The effective max batch size must be <= {_WS_MAX_BATCH_SIZE}"
         )
     refs = [{"wsid": workspace_id, "name": name} for name in obj_names]
-    res = ws.get_object_info3(
-        {"objects": refs, "ignoreErrors": 1, "includeMetadata": 1}
-    )
-    uploaded_obj_names_batch = [
-        info[1]
-        for info in res["infos"]
-        if info is not None and "load_id" in info[10] and info[10]["load_id"] == load_id
-    ]
-    uploaded_obj_upas_batch = [
-        path[0].replace("/", "_") for path in res["paths"] if path is not None
-    ]
-    return uploaded_obj_names_batch, uploaded_obj_upas_batch
+    res = ws.get_object_info3({"objects": refs, "ignoreErrors": 1, "includeMetadata": 1})
+    uploaded_objs_info = [info for info in res["infos"] if info and info[10].get("load_id") == load_id]
+    if not uploaded_objs_info:
+        return list(), list(), list(), list()
+    uploaded_obj_names_batch = [info[1] for info in uploaded_objs_info]
+    uploaded_obj_upas_batch = ["{6}_{0}_{4}".format(*info) for info in uploaded_objs_info]
+
+    if assembly_objs_only:
+        _check_obj_type(workspace_id, load_id, uploaded_objs_info, {loader_common_names.OBJECTS_NAME_ASSEMBLY})
+        assembly_objs_info = uploaded_objs_info
+        genome_objs_info = list()
+    else:
+        _check_obj_type(workspace_id, load_id, uploaded_objs_info, {loader_common_names.OBJECTS_NAME_GENOME})
+        genome_objs_info = uploaded_objs_info
+        try:
+            assembly_objs_ref = [info[10]['Assembly Object'] for info in genome_objs_info]
+        except KeyError:
+            raise ValueError(
+                f"Genome objects must have an 'Assembly Object' field in their metadata"
+            )
+        assembly_objs_spec = [{"ref": ref} for ref in assembly_objs_ref]
+        assembly_objs_info = ws.get_object_info3({"objects": assembly_objs_spec, "includeMetadata": 1})["infos"]
+
+    return uploaded_obj_names_batch, uploaded_obj_upas_batch, assembly_objs_info, genome_objs_info
```

Comment:
Returning 4 things seems excessive, but since it's an internal method only...

Reply:
yea. next PR will fix this.

Reply:
:+1: